server/controllers/entities/lib/patches.js

```diff
@@ -63,12 +63,38 @@ module.exports = {
   byRedirectUri: db.viewByKey.bind(null, 'byRedirectUri'),
 
   create: async params => {
-    const patch = Patch.create(params)
-    return db.postAndReturn(patch)
+    const { currentDoc, updatedDoc, userId } = params
+    const newPatch = Patch.create(params)
+    if (currentDoc.version > 1) {
+      const [ previousPatch ] = await patches_.getLastPatches(currentDoc._id)
+      if (previousPatch && previousPatch.user === userId) {
+        const beforeLastPatch = Patch.revert(currentDoc, previousPatch)
+        const aggregatedPatch = Patch.getDiff(beforeLastPatch, updatedDoc)
+        if (previousPatch.context == null && previousPatch.batch == null) {
+          if (aggregatedPatch.length === 0) {
+            await db.delete(previousPatch._id, previousPatch._rev)
+            return
+          } else if (aggregatedPatch.length < (previousPatch.patch.length + newPatch.patch.length)) {
+            previousPatch.patch = aggregatedPatch
+            return db.putAndReturn(previousPatch)
+          }
```

Comment:
this went from 2 lines to 5 `if` statements. Would it be possible to extract some logic to make it more readable ?

Reply:
I gave it a try in 640766a9, but couldn't really extract much, only try to make it less indigest